app/models/validations/soft_validations.rb

```diff
@@ -28,15 +28,20 @@ def net_income_in_soft_min_range?
   def rent_in_soft_min_range?
     return unless brent && weekly_value(brent) && startdate
 
-    rent_range = LaRentRange.find_by(start_year: collection_start_year, la:, beds:, lettype: get_lettype)
+    rent_range = LaRentRange.find_by(start_year: collection_start_year, la:, beds: validation_beds, lettype: get_lettype)
     rent_range.present? && weekly_value(brent).between?(rent_range.hard_min, rent_range.soft_min)
   end
 
   def rent_in_soft_max_range?
     return unless brent && weekly_value(brent) && startdate
 
-    rent_range = LaRentRange.find_by(start_year: collection_start_year, la:, beds:, lettype: get_lettype)
-    rent_range.present? && weekly_value(brent).between?(rent_range.soft_max, rent_range.hard_max)
+    rent_range = LaRentRange.find_by(start_year: collection_start_year, la:, beds: validation_beds, lettype: get_lettype)
+    max_beds_in_rent_ranges = 4
+    if beds.present? && beds > max_beds_in_rent_ranges
+      rent_range.present? && weekly_value(brent) > rent_range.soft_max
+    else
+      rent_range.present? && weekly_value(brent).between?(rent_range.soft_max, rent_range.hard_max)
+    end
```

Comment:
```suggestion
    if beds.present? && rent_range.present? && beds > LaRentRange::MAX_BEDS
      weekly_value(brent) > rent_range.soft_max
    else
      weekly_value(brent).between?(rent_range.soft_max, rent_range.hard_max)
    end
```

You can move `rent_range.present?` to the first condition as it is shared by both branches.

Reply:
Won't this give an error in the else section if `rent_range` is not present when we try to access `.soft_max` or `.soft_min`?

Reply:
Ah yes! You're right. I think just the `LaRentRange::MAX_BEDS` change, then.